x-pack/plugins/security_solution/public/flyout/right/components/reason.tsx

```diff
@@ -6,49 +6,94 @@
  */
 
 import type { FC } from 'react';
-import React, { useMemo } from 'react';
-import { EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
-import { REASON_DETAILS_TEST_ID, REASON_TITLE_TEST_ID } from './test_ids';
-import { ALERT_REASON_TITLE, DOCUMENT_REASON_TITLE } from './translations';
+import React, { useCallback, useMemo } from 'react';
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
+import { useExpandableFlyoutContext } from '@kbn/expandable-flyout';
+import { getField } from '../../shared/utils';
+import type { PreviewPanelProps } from '../../preview';
+import { PreviewPanelKey } from '../../preview';
+import {
+  REASON_DETAILS_PREVIEW_BUTTON_TEST_ID,
+  REASON_DETAILS_TEST_ID,
+  REASON_TITLE_TEST_ID,
+} from './test_ids';
+import {
+  ALERT_REASON_DETAILS_TEXT,
+  ALERT_REASON_TITLE,
+  DOCUMENT_REASON_TITLE,
+  PREVIEW_ALERT_REASON_DETAILS,
+} from './translations';
 import { useBasicDataFromDetailsData } from '../../../timelines/components/side_panel/event_details/helpers';
-import { defaultRowRenderers } from '../../../timelines/components/timeline/body/renderers';
-import { getRowRenderer } from '../../../timelines/components/timeline/body/renderers/get_row_renderer';
 import { useRightPanelContext } from '../context';
 
 /**
  * Displays the information provided by the rowRenderer. Supports multiple types of documents.
  */
 export const Reason: FC = () => {
-  const { dataAsNestedObject, dataFormattedForFieldBrowser } = useRightPanelContext();
+  const { eventId, indexName, scopeId, dataFormattedForFieldBrowser, getFieldsData } =
+    useRightPanelContext();
   const { isAlert } = useBasicDataFromDetailsData(dataFormattedForFieldBrowser);
+  const alertReason = getField(getFieldsData('kibana.alert.reason'));
```

Comment:
Just a heads up, there are constants for these fields here: `import { ALERT_REASON } from '@kbn/rule-data-utils'`;